ksql-streams/src/main/java/io/confluent/ksql/execution/streams/AggregateParams.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.streams;
+
+import com.google.common.collect.ImmutableMap;
+import io.confluent.ksql.execution.expression.tree.FunctionCall;
+import io.confluent.ksql.execution.function.TableAggregationFunction;
+import io.confluent.ksql.execution.function.UdafUtil;
+import io.confluent.ksql.execution.function.udaf.KudafAggregator;
+import io.confluent.ksql.execution.function.udaf.KudafInitializer;
+import io.confluent.ksql.execution.function.udaf.KudafUndoAggregator;
+import io.confluent.ksql.execution.function.udaf.window.WindowSelectMapper;
+import io.confluent.ksql.function.FunctionRegistry;
+import io.confluent.ksql.function.KsqlAggregateFunction;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public final class AggregateParams {
+  private final KudafInitializer initializer;
+  private final int initialUdafIndex;
+  private final Map<Integer, KsqlAggregateFunction> indexToFunction;
```

Comment:
Rather than a `Map` I would suggest an ordered `List< KsqlAggregateFunction>`.

See my PR: https://github.com/confluentinc/ksql/pull/3409

The indexes in the map are always sequential, starting at `initialUdafIndex`.

Reply:
Agreed. Was trying (and failing) to keep this diff as small as possible :)